web/src/cypress/support/calendar-subscription.ts

```diff
@@ -0,0 +1,164 @@
+import { Chance } from 'chance'
+
+const c = new Chance()
+
+declare global {
+  namespace Cypress {
+    interface Chainable {
+      /**
+       * Creates a new calendar subscription.
+       */
+      createCalendarSubscription: typeof createCalendarSubscription
+
+      /**
+       * Creates an amount of random calendar subscriptions given
+       * by a specified count.
+       */
+      createManyCalendarSubscriptions: typeof createManyCalendarSubscriptions
+
+      /**
+       * Deletes all calendar subscriptions given the specified user ID.
+       * Will default to deleting all subscriptions for 'profile' if no ID provided.
+       */
+      cleanupCalendarSubscriptions: typeof cleanupCalendarSubscriptions
+    }
+  }
+
+  interface CalendarSubscription {
+    id: string
+    name: string
+    reminderMinutes: Array<number>
+    scheduleID: string
+    schedule: Schedule
+    lastAccess: string
+    disabled: boolean
+  }
+
+  interface CalendarSubscriptionOptions {
+    name?: string
+    reminderMinutes?: Array<number>
+    scheduleID?: string
+    disabled?: boolean
+  }
```

Comment:
For the purposes of these tests if a scheduleID wasn't provided I have it create a schedule with random values– I thought that providing a full suite of options was a bit overkill for the current goal

Reply:
I'd add it just so that all API's match. No reason to have "some do, some don't" Everything that takes an optional `scheduleID` takes optional `schedule` currently. Only need to pass it as the sole argument to `createSchedule`, empty or not.

Reply:
No worries, I can add. I just didn’t want to have more code to review (I also had a deleteSubscription method that I eventually deleted). Sounds good though